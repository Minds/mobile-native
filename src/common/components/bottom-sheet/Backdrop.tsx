import React, { FunctionComponent } from 'react';
import {
  BottomSheetBackdrop,
  BottomSheetBackdropProps,
} from '@gorhom/bottom-sheet';
import {
  useAnimatedReaction,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated';

const OPEN_ANIMATION_DURATION = 300;

// fixes flickering https://github.com/gorhom/react-native-bottom-sheet/issues/436
const Backdrop: FunctionComponent<BottomSheetBackdropProps> = ({
  animatedIndex,
  ...rest
}) => {
  const adjustedAnimatedIndex = useSharedValue(0);
  const hasOpened = useSharedValue(false);

  useAnimatedReaction(
    () => animatedIndex.value,
    (data, prev) => {
      if (prev == null) {
        adjustedAnimatedIndex.value = withTiming(
          1,
          { duration: OPEN_ANIMATION_DURATION },
          isFinished => {
            if (isFinished) hasOpened.value = true;
          },
        );
      }

      if (hasOpened.value) adjustedAnimatedIndex.value = data;
    },
  );

  return (
    <BottomSheetBackdrop animatedIndex={adjustedAnimatedIndex} {...rest} />
  );
};

export default Backdrop;
