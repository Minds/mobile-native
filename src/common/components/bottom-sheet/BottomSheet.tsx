import React, { forwardRef, useCallback } from 'react';
import {
  BottomSheetModal,
  BottomSheetModalProps,
  useBottomSheetDynamicSnapPoints,
  BottomSheetBackdrop,
} from '@gorhom/bottom-sheet';
import { StatusBar, View } from 'react-native';
import ThemedStyles, { useStyle } from '../../../styles/ThemedStyles';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import Handle from './Handle';
import MText from '../MText';

interface PropsType extends Omit<BottomSheetModalProps, 'snapPoints'> {
  title?: string;
  subtitle?: string;
  detail?: string;
  autoShow?: boolean;
  snapPoints?: Array<number | string>;
  forceHeight?: number;
}

export default forwardRef<BottomSheetModal, PropsType>((props, ref) => {
  const { title, detail, snapPoints, autoShow, children, ...other } = props;

  const insets = useSafeAreaInsets();

  const snapPointsMemo = React.useMemo(() => snapPoints || ['CONTENT_HEIGHT'], [
    snapPoints,
  ]);

  const {
    animatedHandleHeight,
    animatedSnapPoints,
    animatedContentHeight,
    handleContentLayout,
  } = useBottomSheetDynamicSnapPoints(snapPointsMemo);

  const contStyle = useStyle(styles.contentContainer, {
    paddingBottom: insets.bottom || 24,
  });

  React.useEffect(() => {
    //@ts-ignore
    if (ref && ref.current && autoShow) {
      //@ts-ignore
      ref.current.present();
    }
  }, [autoShow, ref]);

  const renderHandle = useCallback(() => <Handle />, []);

  const renderBackdrop = useCallback(
    props => (
      <BottomSheetBackdrop
        {...props}
        pressBehavior="close"
        opacity={0.5}
        appearsOnIndex={0}
        disappearsOnIndex={-1}
      />
    ),
    [],
  );

  return (
    <BottomSheetModal
      ref={ref}
      topInset={StatusBar.currentHeight || 0}
      handleComponent={renderHandle}
      snapPoints={animatedSnapPoints}
      handleHeight={animatedHandleHeight}
      contentHeight={animatedContentHeight}
      backdropComponent={renderBackdrop}
      enablePanDownToClose={true}
      backgroundComponent={null}
      style={styles.sheetContainer as any}
      {...other}>
      <View style={contStyle} onLayout={handleContentLayout}>
        {Boolean(title) && <MText style={styles.title}>{title}</MText>}
        {Boolean(detail) && <MText style={styles.detail}>{detail}</MText>}
        {children}
      </View>
    </BottomSheetModal>
  );
});

const styles = ThemedStyles.create({
  contentContainer: ['bgPrimaryBackgroundHighlight'],
  title: ['fontXXL', 'bold', 'textCenter', 'marginVertical3x'],
  detail: [
    'fontL',
    'fontMedium',
    'textCenter',
    'marginTop3x',
    'marginBottom5x',
    'colorSecondaryText',
  ],
  sheetContainer: [
    'shadowBlack',
    {
      shadowOffset: {
        width: 0,
        height: -3,
      },
      shadowOpacity: 0.58,
      shadowRadius: 4.0,
      elevation: 16,
    },
  ],
});
