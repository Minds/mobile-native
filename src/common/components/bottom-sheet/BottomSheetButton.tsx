import React from 'react';
import { ViewStyle } from 'react-native';
import { Button } from '~ui';

interface PropsType {
  text: string;
  action?: boolean;
  warning?: boolean;
  onPress?: () => void;
  testID?: string;
  loading?: boolean;
  solid?: boolean;
  disabled?: boolean;
  containerStyle?: ViewStyle | ViewStyle[];
}

export default function BottomSheetButton({
  text,
  onPress,
  action,
  warning,
  loading,
  testID,
  solid,
  containerStyle,
}: PropsType) {
  return (
    <Button
      top="L"
      horizontal="L"
      onPress={onPress}
      testID={testID}
      loading={loading}
      mode={action || warning ? (solid ? 'solid' : 'outline') : 'solid'}
      type={action ? 'action' : warning ? 'warning' : undefined}
      containerStyle={containerStyle}>
      {text}
    </Button>
  );
}
