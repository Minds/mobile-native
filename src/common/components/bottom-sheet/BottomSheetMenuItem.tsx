import React from 'react';
import { View } from 'react-native';
import { Icon, IconType } from 'react-native-elements';
import ThemedStyles, { useStyle } from '../../../styles/ThemedStyles';
import MPressable from '../MPressable';
import MText from '../MText';
import { B1 } from '~ui';
import { BottomSheetMethods } from '@gorhom/bottom-sheet/lib/typescript/types';

export type BottomSheetMenuItemProps = {
  title: string | JSX.Element;
  iconName?: string;
  iconType?: IconType;
  iconSize?: number;
  testID?: string;
  onPress?: (ref?: BottomSheetMethods) => void;
  style?: any;
  textStyle?: any;
};

const BottomSheetMenuItem = ({
  iconName,
  iconType,
  title,
  onPress,
  iconSize,
  style,
  textStyle,
  testID = '',
}: BottomSheetMenuItemProps) => {
  iconSize = iconSize || 24;

  const txtStyle = useStyle(styles.menuText, textStyle);
  const containerStyle = useStyle(styles.menuContainer, style);

  return (
    <MPressable
      style={containerStyle}
      onPress={onPress}
      testID={testID}
      disabled={!onPress}>
      {Boolean(iconName) && (
        <View style={styles.iconContainer}>
          <Icon
            size={iconSize}
            name={iconName!}
            type={iconType}
            iconStyle={ThemedStyles.style.colorSecondaryText}
          />
        </View>
      )}
      {textStyle ? (
        <MText style={txtStyle}>{title}</MText>
      ) : (
        <B1 font="medium">{title}</B1>
      )}
    </MPressable>
  );
};

export default BottomSheetMenuItem;

const styles = ThemedStyles.create({
  iconContainer: [{ width: 25 }, 'alignCenter', 'marginRight4x'],
  menuContainer: [
    'alignCenter',
    'rowJustifyStart',
    'paddingVertical3x',
    'paddingHorizontal4x',
  ],
  menuText: ['fontL', 'fontMedium'],
});
