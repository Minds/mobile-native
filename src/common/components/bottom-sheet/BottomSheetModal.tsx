import React, { forwardRef, useCallback } from 'react';
import {
  BottomSheetModal,
  BottomSheetModalProps,
  useBottomSheetDynamicSnapPoints,
  BottomSheetBackdrop,
} from '@gorhom/bottom-sheet';
import { StatusBar, View } from 'react-native';
import ThemedStyles, { useStyle } from '../../../styles/ThemedStyles';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import Handle from './Handle';
import useBackHandler from './useBackHandler';
import { H3, B1, IconButton } from '~ui';
import { HORIZONTAL } from '~/styles/Tokens';

interface PropsType extends Omit<BottomSheetModalProps, 'snapPoints'> {
  title?: string;
  subtitle?: string;
  detail?: string;
  autoShow?: boolean;
  snapPoints?: Array<number | string>;
  forceHeight?: number;
  onBack?: () => void;
  onDismiss?: () => void;
}

const MBottomSheetModal = forwardRef<BottomSheetModal, PropsType>(
  (props, ref) => {
    const {
      title,
      detail,
      snapPoints,
      autoShow,
      onBack,
      onDismiss,
      children,
      ...other
    } = props;
    const { onAnimateHandler } = useBackHandler(
      // @ts-ignore
      useCallback(() => ref?.current?.close(), [ref]),
      props,
    );

    const insets = useSafeAreaInsets();

    const snapPointsMemo = React.useMemo(
      () => snapPoints || ['CONTENT_HEIGHT'],
      [snapPoints],
    );

    const {
      animatedHandleHeight,
      animatedSnapPoints,
      animatedContentHeight,
      handleContentLayout,
    } = useBottomSheetDynamicSnapPoints(snapPointsMemo);

    const contStyle = useStyle(styles.contentContainer, {
      paddingBottom: insets.bottom || 24,
    });

    React.useEffect(() => {
      //@ts-ignore
      if (ref && ref.current && autoShow) {
        //@ts-ignore
        ref.current.present();
      }
    }, [autoShow, ref]);

    const renderHandle = useCallback(() => <Handle />, []);

    const renderBackdrop = useCallback(
      backdropProps => (
        <BottomSheetBackdrop
          {...backdropProps}
          pressBehavior="close"
          opacity={0.5}
          appearsOnIndex={0}
          disappearsOnIndex={-1}
        />
      ),
      [],
    );

    return (
      <BottomSheetModal
        ref={ref}
        topInset={StatusBar.currentHeight || 0}
        handleComponent={renderHandle}
        snapPoints={animatedSnapPoints}
        handleHeight={animatedHandleHeight}
        contentHeight={animatedContentHeight}
        backdropComponent={renderBackdrop}
        enablePanDownToClose={true}
        backgroundComponent={null}
        style={styles.sheetContainer as any}
        {...other}
        onAnimate={onAnimateHandler}>
        <View style={contStyle} onLayout={handleContentLayout}>
          {Boolean(title) && (
            <View style={styles.header}>
              <H3 vertical="M" align="center">
                {title}
              </H3>
              {Boolean(onBack) && (
                <View style={styles.backIcon}>
                  <IconButton
                    size="large"
                    name="chevron-left"
                    onPress={onBack}
                  />
                </View>
              )}
              {Boolean(onDismiss) && (
                <View style={styles.closeIcon}>
                  <IconButton size="large" name="close" onPress={onDismiss} />
                </View>
              )}
            </View>
          )}

          {Boolean(detail) && (
            <B1
              horizontal="L"
              bottom="M"
              align="center"
              color="secondary"
              font="medium">
              {detail}
            </B1>
          )}
          {children}
        </View>
      </BottomSheetModal>
    );
  },
);

const styles = ThemedStyles.create({
  contentContainer: ['bgPrimaryBackgroundHighlight'],
  title: ['fontXXL', 'bold', 'textCenter', 'marginVertical3x'],
  detail: [
    'fontL',
    'fontMedium',
    'textCenter',
    'marginTop3x',
    'marginBottom5x',
    'colorSecondaryText',
  ],
  sheetContainer: [
    'shadowBlack',
    {
      shadowOffset: {
        width: 0,
        height: -3,
      },
      shadowOpacity: 0.58,
      shadowRadius: 4.0,
      elevation: 16,
    },
  ],
  header: [
    {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      paddingHorizontal: HORIZONTAL,
    },
    'paddingVertical2x',
  ],
  backIcon: [
    {
      position: 'absolute',
      left: HORIZONTAL,
    },
  ],
  closeIcon: [
    {
      position: 'absolute',
      right: HORIZONTAL,
    },
  ],
});

export default MBottomSheetModal;
