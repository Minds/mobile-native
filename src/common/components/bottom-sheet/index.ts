import BottomSheet, { PropsType } from './BottomSheet';
import BottomSheetModal, { BottomSheetModalHandle } from './BottomSheetModal';
import BottomSheetButton from './BottomSheetButton';
import BottomSheetMenuItem, {
  BottomSheetMenuItemProps,
} from './BottomSheetMenuItem';
import RadioButton from './RadioButton';
import SectionTitle from './SectionTitle';
import { pushBottomSheet } from './BottomSheetScreen';

export {
  BottomSheet,
  BottomSheetModal,
  BottomSheetMenuItem,
  RadioButton,
  SectionTitle,
  BottomSheetButton,
  pushBottomSheet,
};

export type {
  BottomSheetModalHandle,
  BottomSheetMenuItemProps,
  PropsType as BottomSheetProps,
};
