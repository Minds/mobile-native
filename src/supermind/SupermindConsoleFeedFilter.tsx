import React from 'react';
import { TextStyle, ViewStyle } from 'react-native';
import { observer } from 'mobx-react';
import { useBottomSheetModal } from '@gorhom/bottom-sheet';

import { SectionTitle, RadioButton } from '~/common/components/bottom-sheet';
import { Spacer } from '~ui';
import BaseFeedFilter from '~/common/components/feed-filters/BaseFeedFilter';
import i18nService from '~/common/services/i18n.service';

export type SupermindFilterType = 'pending' | 'accepted' | 'expired' | 'failed';

type PropsType = {
  hideLabel?: boolean;
  onFilterChange: (value: SupermindFilterType) => void;
  value: SupermindFilterType;
  containerStyles?: ViewStyle | ViewStyle[];
  textStyle?: TextStyle | TextStyle[];
};

/**
 * Supermind Feed Filter selector
 */
const SupermindConsoleFeedFilter = ({
  onFilterChange,
  value,
  ...otherProps
}: PropsType) => {
  const { dismiss } = useBottomSheetModal();

  const onChange = v => {
    onFilterChange(v);
    dismiss();
  };

  return (
    <BaseFeedFilter
      label={i18nService.t(`supermind.filter.${value}`)}
      {...otherProps}>
      <SectionTitle>Filter Supermind Offers</SectionTitle>
      <RadioButton
        testID="PendingRadio"
        title="Pending"
        onPress={() => onChange('pending')}
        selected={value === 'pending'}
      />
      <RadioButton
        title="Accepted"
        testID="AcceptedRadio"
        onPress={() => onChange('accepted')}
        selected={value === 'accepted'}
      />
      <RadioButton
        title="Expired"
        testID="ExpiredRadio"
        onPress={() => onChange('expired')}
        selected={value === 'expired'}
      />
      <RadioButton
        title="Failed"
        testID="FailedRadio"
        onPress={() => onChange('failed')}
        selected={value === 'failed'}
      />
      <Spacer top="L" />
    </BaseFeedFilter>
  );
};

export default observer(SupermindConsoleFeedFilter);
