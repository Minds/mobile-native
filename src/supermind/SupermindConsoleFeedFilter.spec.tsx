import * as React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import SupermindConsoleFeedFilter, {
  SupermindFilterType,
} from './SupermindConsoleFeedFilter';
import { BottomSheetModalProvider } from '@gorhom/bottom-sheet';

describe('SupermindFeedFilter', () => {
  it('should render', async () => {
    render(
      <BottomSheetModalProvider>
        <SupermindConsoleFeedFilter
          onFilterChange={v => console.log(v)}
          value="accepted"
        />
      </BottomSheetModalProvider>,
    );

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('It should fire onChange', async () => {
    let current: SupermindFilterType = 'accepted';

    const onFilterChange = jest.fn();

    render(
      <BottomSheetModalProvider>
        <SupermindConsoleFeedFilter
          onFilterChange={onFilterChange}
          value={current}
        />
      </BottomSheetModalProvider>,
    );

    // press accepted
    fireEvent.press(screen.getByTestId('AcceptedRadio'));

    expect(onFilterChange).toBeCalledWith('accepted');

    // press failed
    fireEvent.press(screen.getByTestId('FailedRadio'));

    expect(onFilterChange).toBeCalledWith('failed');

    // press expired
    fireEvent.press(screen.getByTestId('ExpiredRadio'));

    expect(onFilterChange).toBeCalledWith('expired');

    // press pending
    fireEvent.press(screen.getByTestId('PendingRadio'));

    expect(onFilterChange).toBeCalledWith('pending');
  });
});
