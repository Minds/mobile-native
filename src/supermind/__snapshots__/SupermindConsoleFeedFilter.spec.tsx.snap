// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SupermindFeedFilter should render 1`] = `
Array [
  <View
    onPress={[Function]}
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        },
        undefined,
      ]
    }
    testID="FilterToggle"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 18,
          },
          Object {
            "color": "#AEB0B8",
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰈶
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
          },
          Object {
            "fontFamily": "Roboto",
          },
          Array [
            Object {
              "paddingLeft": 4,
            },
            Object {
              "fontSize": 15,
            },
            undefined,
          ],
        ]
      }
    >
      Accepted
    </Text>
  </View>,
  <Modal
    backdropComponent={[Function]}
    backgroundComponent={null}
    contentHeight={null}
    enablePanDownToClose={true}
    handleComponent={[Function]}
    handleHeight={null}
    hardwareAccelerated={false}
    onAnimate={[Function]}
    snapPoints={null}
    stackBehavior="push"
    style={
      Array [
        Object {
          "shadowColor": "#000000",
        },
        Object {
          "elevation": 16,
          "shadowOffset": Object {
            "height": -3,
            "width": 0,
          },
          "shadowOpacity": 0.58,
          "shadowRadius": 4,
        },
      ]
    }
    topInset={0}
    visible={true}
  >
    <View
      onLayout={null}
      style={
        Array [
          Array [
            Object {
              "backgroundColor": "#1E242A",
            },
          ],
          Object {
            "paddingBottom": 24,
          },
        ]
      }
    >
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "marginBottom": 16,
                "marginLeft": 20,
                "marginTop": 16,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {
                    "fontFamily": "Roboto-Medium",
                  },
                  Object {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "textAlign": "left",
                  },
                ]
              }
            >
              Filter Supermind Offers
            </Text>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                },
                Array [
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                    },
                    Object {
                      "paddingVertical": 12,
                    },
                    Object {
                      "paddingHorizontal": 16,
                    },
                  ],
                  undefined,
                ],
              ]
            }
            testID="PendingRadio"
          >
            <View
              style={
                Array [
                  Object {
                    "width": 25,
                  },
                  Object {
                    "alignItems": "center",
                  },
                  Object {
                    "marginRight": 16,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#242424",
                            "fontSize": 24,
                          },
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#B8C1CA",
                          },
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {
                    "fontFamily": "Roboto-Medium",
                  },
                  Object {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "textAlign": "left",
                  },
                ]
              }
            >
              Pending
            </Text>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                },
                Array [
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                    },
                    Object {
                      "paddingVertical": 12,
                    },
                    Object {
                      "paddingHorizontal": 16,
                    },
                  ],
                  undefined,
                ],
              ]
            }
            testID="AcceptedRadio"
          >
            <View
              style={
                Array [
                  Object {
                    "width": 25,
                  },
                  Object {
                    "alignItems": "center",
                  },
                  Object {
                    "marginRight": 16,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#242424",
                            "fontSize": 24,
                          },
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#B8C1CA",
                          },
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {
                    "fontFamily": "Roboto-Medium",
                  },
                  Object {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "textAlign": "left",
                  },
                ]
              }
            >
              Accepted
            </Text>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                },
                Array [
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                    },
                    Object {
                      "paddingVertical": 12,
                    },
                    Object {
                      "paddingHorizontal": 16,
                    },
                  ],
                  undefined,
                ],
              ]
            }
            testID="ExpiredRadio"
          >
            <View
              style={
                Array [
                  Object {
                    "width": 25,
                  },
                  Object {
                    "alignItems": "center",
                  },
                  Object {
                    "marginRight": 16,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#242424",
                            "fontSize": 24,
                          },
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#B8C1CA",
                          },
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {
                    "fontFamily": "Roboto-Medium",
                  },
                  Object {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "textAlign": "left",
                  },
                ]
              }
            >
              Expired
            </Text>
          </View>
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                },
                Array [
                  Array [
                    Object {
                      "alignItems": "center",
                    },
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "flex-start",
                    },
                    Object {
                      "paddingVertical": 12,
                    },
                    Object {
                      "paddingHorizontal": 16,
                    },
                  ],
                  undefined,
                ],
              ]
            }
            testID="FailedRadio"
          >
            <View
              style={
                Array [
                  Object {
                    "width": 25,
                  },
                  Object {
                    "alignItems": "center",
                  },
                  Object {
                    "marginRight": 16,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#242424",
                            "fontSize": 24,
                          },
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#B8C1CA",
                          },
                          Object {
                            "fontFamily": "Ionicons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                  },
                  Object {
                    "fontFamily": "Roboto-Medium",
                  },
                  Object {
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  Object {
                    "textAlign": "left",
                  },
                ]
              }
            >
              Failed
            </Text>
          </View>
          <View
            style={
              Object {
                "marginTop": 16,
              }
            }
          />
          <View
            style={
              Object {
                "marginHorizontal": 16,
                "marginTop": 16,
              }
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderRadius": 100,
                      "flexDirection": "column",
                      "justifyContent": "center",
                      "overflow": "hidden",
                    },
                    undefined,
                    Object {
                      "minHeight": 42,
                      "paddingHorizontal": 24,
                      "paddingVertical": 2,
                    },
                    undefined,
                    undefined,
                    undefined,
                    false,
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#3f4b56",
                      "borderColor": "transparent",
                      "borderRadius": 100,
                      "borderWidth": 1,
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, 0.1)",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  collapsable={false}
                  style={
                    Object {
                      "transform": Array [
                        Object {
                          "scale": 1,
                        },
                      ],
                    }
                  }
                >
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#FFFFFF",
                        },
                        Object {
                          "fontFamily": "Roboto-Medium",
                        },
                        Object {
                          "fontSize": 18,
                          "lineHeight": 24,
                        },
                        Object {
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    Close
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </Modal>,
]
`;
